pkg/apitype/stacks.go

```diff
@@ -53,6 +56,11 @@ type EncryptValueRequest struct {
 	Plaintext []byte `json:"plaintext"`
 }
 
+type CurrentUpdateStatus struct {
+	CurrentUpdateStarted int64 `json:"currentUpdateStarted,omitempty"`
+	CurrentUpdateAuthor string `json:"currentUpdateAuthor,omitempty"`
```

Comment:
Let us not add the `,omitempty` encoding specifier here. The rule of thumb is "if the type is a pointer to something, pass `omitempty`, otherwise do not add it.

If you are unfamiliar with what "omitempty" does, there are a few (poorly written) blog posts that can give you an idea: https://vsupalov.com/go-json-omitempty/.

But the reason it is important to not include it here is that it makes using the `apitype` from the frontend more difficult.

If for whatever reason the `CurrentUpdateAuthor` field weren't set, and the default value of `""` were serialized to JSON, the field would be missing entirely. This means that any JavaScript code consuming the JSON type would need to check for `undefined` instead of just receiving the `""` value.

For our REST API type definitions, we want to err on the side of being as explicit as possible -- i.e. sending empty strings and 0s across the wire.

The exception however is when dealing with Go values that could be `nil`. (i.e. anything that is a pointer, `map`, etc.) In those cases, we sometimes do specify the `omitempty`. The rational is that the JavaScript client is already aware that the field may not be sent, and therefore isn't any harder to use. And omitting the field from the JSON output saves XX bytes, which slowly adds up.

e.g. when you added a field to `StackSummary`, using `omitempty` here is good:
`CurrentUpdate *CurrentUpdateStatus `json:"currentUpdate,omitempty"`, because the JSON result will just return `undefined` when looking for the key `"currentUpdate"`. However, if we have the JSON result of `CurrentUpdateStatus`, we want both `currentUpdateAuthor` and `currentUpdateStarted` to have a value. (Even if 0 or "".)